agent/util-scripts/modules/metaclass.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+import json
+import os
+
+
+class ToolMetadata:
+    def __init__(self, mode, context, logger):
+        self.logger = logger
+        if mode not in ("redis", "json"):
+            raise Exception()
+        if not context:
+            raise Exception()
+        self.mode = mode
+        if mode == "redis":
+            self.redis_server = context
+            self.json_file = None
+        else:
+            if not self.mode == "json":
```

Comment:
If `mode not in ("redis", "json"):` above makes this unnecessary. Or, alternatively, if we really want to say "either redis or it doesn't matter and we read json anyway" then maybe it's the earlier check that's unnecessary! (But I'd prefer to keep the Exception and just drop this redundant check. I don't really like the implications, anyway.)